scalac-scoverage-runtime/jvm/src/test/scala/scoverage/InvokerUseEnvironmentTest.scala

```diff
@@ -0,0 +1,80 @@
+package scoverage
+
+import java.nio.file.{Files, Paths}
+import org.scalatest.{BeforeAndAfter, FunSuite}
+import scoverage.Platform.File
+
+/**
+ * Verify that [[Invoker.invokedWriteToClasspath()]] works as expected.
+ */
+class InvokerUseEnvironmentTest extends FunSuite with BeforeAndAfter {
+
+
+  val instrumentsDir = Array(
+    new File("target/invoker-test.instrument0"),
+    new File("target/invoker-test.instrument1")
+  )
+
+  before {
+    deleteInstrumentFiles()
+    instrumentsDir.foreach(_.mkdirs())
+    createNewInstruments()
+  }
+
+  test("calling Invoker.invokedUseEnvironments puts the data on env var SCOVERAGE_MEASUREMENT_PATH" +
+    " and copies coverage files from instruments directory.") {
+
+    val testIds: Set[Int] = (1 to 10).toSet
+
+    testIds.map { i: Int => Invoker.invokedWriteToClasspath(i, instrumentsDir(i % 2).toString)}
+
+    // Verify measurements went to correct directory under the environment variable.
+    val dir0 = s"${System.getenv("SCOVERAGE_MEASUREMENT_PATH")}/${Invoker.safe_name(instrumentsDir(0).toString)}"
+    val measurementFiles3 = Invoker.findMeasurementFiles(dir0)
+    val idsFromFile3 = Invoker.invoked(measurementFiles3.toIndexedSeq)
+    assert (idsFromFile3 == testIds.filter { i: Int => i % 2 == 0})
+
+
+    val dir1 = s"${System.getenv("SCOVERAGE_MEASUREMENT_PATH")}/${Invoker.safe_name(instrumentsDir(1).toString)}"
+    val measurementFiles4 = Invoker.findMeasurementFiles(dir1)
+    val idsFromFile4 = Invoker.invoked(measurementFiles4.toIndexedSeq)
+    assert (idsFromFile4 == testIds.filter { i: Int => i % 2 == 1})
+
+    // Verify that coverage files have been copied correctly.
+    assert(Files.exists(Paths.get(s"$dir0/scoverage.coverage")))
+    assert(Files.exists(Paths.get(s"$dir1/scoverage.coverage")))
+  }
+
+  after {
+    deleteInstrumentFiles()
+    instrumentsDir.foreach(_.delete())
+    deleteMeasurementFolders()
+  }
+
+  private def deleteInstrumentFiles(): Unit = {
+    instrumentsDir.foreach((md) => {
+      if (md.isDirectory)
+        md.listFiles().foreach(_.delete())
+    })
+  }
+
+  private def deleteMeasurementFolders(): Unit = {
+    val d = s"${System.getenv("SCOVERAGE_MEASUREMENT_PATH")}"
```

Comment:
This test should not be expect this env var to already be set.

What I'd recommend instead is changing the code in the invoker to consume a Java "system property", rather than an env var, and to then set the property during the test. System properties can be changed at runtime (see "Writing system properties" here: https://docs.oracle.com/javase/tutorial/essential/environment/sysprop.html).

This would avoid setting it in the `build.sbt` file, which affects _all_ tests: not just this one.

Reply:
...but before doing this, see the comment below about how the dataDir is initialized. That change would also allow for setting the dataDir in tests.